source/core_http_client.c

```diff
@@ -749,9 +749,16 @@ static int httpParserOnHeadersCompleteCallback( http_parser * pHttpParser )
     pResponse = pParsingContext->pResponse;
 
     assert( pResponse != NULL );
+    assert( pParsingContext->pBufferCur != NULL );
```

Comment:
While reading this code I noticed that the following comment on line #654 seems a bit inaccurate:

```c
    /* If this is the first time httpParserOnHeaderFieldCallback() has been
     * invoked on a response, then the headers location in the response was set
     * to NULL in the preceding httpParseOnStatusFieldCallback(). */
```

Is the headers location set to NULL in httpParseOnStatusFieldCallback or in parseHttpResponse?